android/app/src/main/java/app/organicmaps/routing/NavigationService.java

```diff
@@ -207,11 +220,13 @@ public int onStartCommand(@NonNull Intent intent, int flags, int startId)
       startForeground(NavigationService.NOTIFICATION_ID, getNotificationBuilder(this).build());
     }
 
-    // Tests on different devices demonstrated that background location works significantly longer when
-    // requested AFTER starting the foreground service. Restarting the location is also necessary to
-    // re-subscribe for more frequent GPS updates for navigation.
-    LocationHelper.from(this).restart();
+    // Subscribe to location updates. This call is idempotent.
+    LocationHelper.from(this).addListener(this);
+
+    // Restart the location with more frequent refresh interval for navigation.
+    LocationHelper.from(this).restartWithNewMode();
 
+    // Please make this service START_STICKY after fixing the issues at the beginning of the function.
```

Comment:
nit:
final LocationHelper lh = LocationHelper.from(this);
lh.addListener(this);
lh.restartWithNewMode();

Put todo here?

Reply:
Filed a ticket https://github.com/organicmaps/organicmaps/issues/6233 and added a link to the code.